### PR TITLE
More rework and cleanups to get closer to the place where we can report bind errors

### PR DESCRIPTION
**Pull request by Luap99**
see commits

### CODE REVIEW

**Comment by Luap99** [comment]
@mheon PTAL This doesn't yet include the error handling we talked about because it seems it doesn't quite work the way I hopped. I likely need to roll our own multi error type for it.

File: src/server/serve.rs [hunk 1]
```diff
@@ -234,12 +207,81 @@ async fn start_dns_server(
...
-        CoreDns::new(addr, port, name, backend, rx, no_proxy).context("new dns server")?;
+    let mut server = CoreDns::new(addr, port, name, backend, rx, no_proxy, nameservers);
     server.run().await.context("run dns server")
 }
 
+async fn read_config_and_spawn(
+    config_path: &str,
+    port: u32,
+    filter_search_domain: &str,
+    handles_v4: &mut ThreadHandleMap<Ipv4Addr>,
```
Comment (mheon) [inline src/server/serve.rs:220]: So this is declared as a HashMap, but counts as a ThreadHandleMap? Fascinating.

Comment (Luap99) [reply src/server/serve.rs:220]: type inference is very good in rust
https://doc.rust-lang.org/rust-by-example/types/inference.html

**Pull request merged**